ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -809,6 +822,10 @@ let migrate_send'  ~__context ~vm ~dest ~live ~vdi_map ~vif_map ~options =
 
   let vdi_map = vdi_map @ extra_vdi_map in
 
+  (* The vdi_map should be complete at this point - it should include all the
+     VDIs of the VM all extra VDIs in the all_vdis list. *)
```

Comment:
This is a bit garbled... did you mean to write `and the extra` rather than `all extra`?

Reply:
Good spot! Yes, I meant something like that.

Reply:
`check_vdi_map vms_vdis` is invoked towards the top of the function, so I'd leave it as it is, because moving it will change the order of checks and the errors that are returned in different scenarios. I think it's important that we verify the VDI map early. However, I could still do a sanity check that these extra VDIs are all defined in the `extra_vdi_map`, and move the `all_vdis` here.

Reply:
Yes, I wasn't suggesting removing the existing `check_vdi_map ~__context vms_vdis vdi_map` that we do further up the function, before adding the extra things to `vdi_map`.
I was suggesting adding a `check_vdi_map ~__context all_vdis vdi_map` at this point in the code.